cartographer/io/fake_file_writer.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/io/fake_file_writer.h"
+
+#include "glog/logging.h"
+#include "gtest/gtest.h"
```

Comment:
@schwoere @wohe why are we including `gtest.h` here? This is the only place in cartographer's codebase where gtest is included in a file which is not related to testing (i.e. not a `_test.cc` or not a test helper). So this gets included in compilation when building the libcartographer target itself. And it's also incorrect - libcartographer is not linked against gmock/gtest, and does not have the include path for gtest set up. This passing CI is probably just pure coincidence, since the header was available from `/usr/include` or wherever. In an environment where gtest is not available in a standard location, this causes a build failure.

Reply:
This fake class is directly used in the fake_file_writer_test and therefore is just used for testing purposes. I updated the bazel BUILD file in this PR as well: All test helpers, mock and fake classes are now built into the cartographer_test_library. I am sending another PR for updating the CMake file now.
As a side note here: Before I updated the BUILD file, CI didn't compile because of gtest being included in fake_file_writer.cc.